### ISSUE TITLE
After setting level in runtime, line number(%l) doesn't work

### ISSUE THREAD

**Issue by anselmo-coolstay** [log4js-node/log4js-node#1215]
Hi. I've installed latest version, v6.4.3.

If logging level is set at runtime, the variable for line number (%l) doesn't work.

### Code

```javascript
'use strict';

const log4js = require('log4js');

const opts = {
  appenders: {
    customized: {
      type: 'stdout',
      layout: {
        type: 'pattern',
        pattern: '%[[%d] [%5p] [%c](%3l)%] %m',
      },
    },
  },
  categories: {
    default: {
      appenders: ['customized'],
      level: 'debug',
      enableCallStack: true,
    },
  },
};

const logger = log4js.configure(opts).getLogger('TEST');

logger.error('before set level');
logger.level = 'info';
logger.error('after set level');
```

### RESULT

```
[2022-03-17T12:22:41.695] [ERROR] [TEST]( 26) before set level
[2022-03-17T12:22:41.698] [ERROR] [TEST](   ) after set level
```

**Comment by peteriman** [comment]
Thanks for the report. I am able to replicate it on `log4js@6.0.0`.
Would you mind sharing what was the previous version that you had it working on?

**Comment by anselmo-coolstay** [comment]
In v6.3.0, it works well.

**Comment by peteriman** [comment]
Yup, I went all the way down to `log4js@6.0.0` and the bug is present there as well.

Thought you would have a working version so I could focus on the changes. But it's OK. I'll try to fix it.

**Comment by peteriman** [comment]
Note to self: 
It seems that setting `logger.level` will reset `logger.useCallStack`.

```javascript
> logger.error('before set level');
[2022-03-17T23:56:28.242] [ERROR] [TEST](  1) before set level
undefined

> logger.useCallStack;
true

> logger.level = 'info';
'info'

> logger.useCallStack;
false

> logger.error('after set level');
[2022-03-17T23:56:28.564] [ERROR] [TEST](   ) after set level
undefined

> logger.useCallStack = true
true

> logger.error('after set useCallStack');
[2022-03-17T23:57:40.910] [ERROR] [TEST](  1) after set useCallStack
undefined
```

**Comment by peteriman** [comment]
Released in `log4js@6.4.4`.

@anselmo-coolstay Please let me know if there are still issues after upgrading.
